cardano-api/src/Cardano/Api/Typed.hs

```diff
@@ -1647,88 +1650,96 @@ toJSONmOfnChecks keys required total
+                             case anyM <|> allM <|> atLeastM of
+                               Just mss -> return mss
+                               Nothing -> fail "No multi-signature script found"
  where
-  convertValueToMultiSigScript :: Value -> Maybe MultiSigScript
+  convertValueToMultiSigScript :: Value -> Aeson.Parser (Maybe MultiSigScript)
   convertValueToMultiSigScript (Object valueHm) =
      payKeyHash' valueHm <|> any' valueHm  <|> all' valueHm <|> atLeast valueHm
-  convertValueToMultiSigScript _ = Nothing
-
-  payKeyHash' :: HMS.HashMap Text Value -> Maybe MultiSigScript
-  payKeyHash' hm = case HMS.lookup "type" hm of
-                    Just (String "requireSignature") ->
-                      case HMS.lookup "paymentKeyHash" hm of
-                        Just (String pkh) -> Just . RequireSignature $ convertToHash pkh
-                        Just val -> error $ "\"paymentKeyHash\" value is not a payment key hash. Value: " <> show val
-                        Nothing -> error "\"paymentKeyHash\" field is empty in your multi-signature script."
-                    _ -> Nothing
-
-  gatherMultiSigScripts :: Vector Value -> [MultiSigScript]
-  gatherMultiSigScripts = catMaybes . Vector.toList . Vector.map convertValueToMultiSigScript
-
-  any' :: HMS.HashMap Text Value -> Maybe MultiSigScript
-  any' hm = case HMS.lookup "type" hm of
-              Just (String "any") -> case HMS.lookup "scripts" hm of
-                                       Just (Array anyValue) ->
-                                         Just . RequireAnyOf $ gatherMultiSigScripts anyValue
-                                       _ -> Nothing
-              _ -> Nothing
-
-  all' :: HMS.HashMap Text Value -> Maybe MultiSigScript
-  all' hm = case HMS.lookup "type" hm of
-             Just (String "all") -> case HMS.lookup "scripts" hm of
-                                      Just (Array allVectorValue) ->
-                                        Just . RequireAllOf $ gatherMultiSigScripts allVectorValue
-                                      _ -> Nothing
-             _ -> Nothing
-
-  atLeast :: HMS.HashMap Text Value -> Maybe MultiSigScript
+  convertValueToMultiSigScript _ = return Nothing
+
+  payKeyHash' :: HMS.HashMap Text Value -> Aeson.Parser (Maybe MultiSigScript)
+  payKeyHash' hm =
+    case HMS.lookup "type" hm of
+      Just (String "requireSignature") ->
+        case HMS.lookup "paymentKeyHash" hm of
+          Just (String pkh) -> Just . RequireSignature <$> convertToHash pkh
+          Just val ->
+            fail $ "\"paymentKeyHash\" value is not a payment key hash. Value: " <> show val
+          Nothing ->
+            fail "\"paymentKeyHash\" field is empty in your multi-signature script."
+      _ -> return Nothing
+
+  gatherMultiSigScripts :: Vector Value -> Aeson.Parser [MultiSigScript]
+  gatherMultiSigScripts vs =
+    catMaybes <$> (sequence . Vector.toList $ Vector.map convertValueToMultiSigScript vs)
```

Comment:
So we should eliminate the `catMaybes` here that ignores elements that fail to parse. Parse failures should be failures.